### ISSUE TITLE
[QUESTION] Examples for helm repo authentication

### ISSUE THREAD

**Issue by jmorsecode** [mgoltzsche/khelm#44]
Would it be possible to elaborate further on the repository authentication usage?  
From https://github.com/mgoltzsche/khelm#repository-configuration:

> Repository credentials can be configured using Helm's repositories.yaml which can be passed through as Secret to generic build jobs. 

The phrasing here implies this is optional but not a khelm default. I understand how a standard helm repositories.yaml functions, but how would this work with a full gitops workflow where the repo credentials are encrypted with KSOPS?

It would seem to be beneficial to have an example added showing how this would be implemented.

**Comment by mgoltzsche** [comment]
> The phrasing here implies this is optional but not a khelm default.

Right, it is optional since khelm allows you to use any repo when [`/helm/repository/repositories.yaml`](https://github.com/mgoltzsche/khelm/blob/v2.2.1/Dockerfile#L4) is not present (default within the container) or `KHELM_TRUST_ANY_REPO=true` is set.

To give an example, imagine you have a file `repositories.yaml` (that you can generate using e.g. `helm repo add`) as follows:
```
apiVersion: ""
generated: "0001-01-01T00:00:00Z"
repositories:
- name: fake
  url: https://example.org/repo/stable
  username: "fake-user"
  password: "fake-password"
```
You can make khelm use it as follows:
```
$ docker run --rm -v "`pwd`/repositories.yaml:/helm/repository/repositories.yaml" mgoltzsche/khelm:2.2.1 template myrelease mychart --repo=https://example.org/repo/stable
Running khelm 2.2.1 (helm 3.10.1)
Using repository "https://example.org/repo/stable" (as user "fake-user")
Downloading repository index of https://example.org/repo/stable
khelm: load chart mychart: download repo index: looks like "https://example.org/repo/stable" is not a valid chart repository or cannot be reached: failed to fetch https://example.org/repo/stable/index.yaml : 404 Not Found
```
While the example above does not actually render a chart since the I specified a fake URL, the log message indicates that khelm uses the credentials from the mounted `repositories.yaml`: `Using repository "https://example.org/repo/stable" (as user "fake-user")`.

However, khelm does not support an encrypted `repositories.yaml` file. When using it within the cluster, it requires you to store your `repositories.yaml` file within a Kubernetes Secret and mount it into the khelm container. When you want to store your `repositories.yaml` encrypted within a git repository and deploy it via GitOps, you could use a Kubernetes controller such as [sealed-secrets](https://github.com/bitnami-labs/sealed-secrets) that decrypts it within the cluster.
Since secret management in git is a cross-cutting concern that is not specific to khelm, I don't think it makes sense to make khelm support it directly but to leverage other tools made specifically for that concern - unless somebody convinces me otherwise.

_(Fwiw, there is an [example showing how to use encrypted Helm values with sops](https://github.com/mgoltzsche/khelm/tree/v2.2.1/example/helm-secrets) but that doesn't apply to the `repositories.yaml`.)_

**Comment by mgoltzsche** [comment]
I just realize that using the sealed-secrets controller in addition to [KSOPS](https://github.com/viaduct-ai/kustomize-sops) makes your setup more complex which I understand nobody wants. I didn't know that tool. Though, since khelm doesn't support decrypting the `repositories.yaml` currently, you'd need to let another tool decrypt it upfront somehow.
How would you imagine this to work ideally?

**Comment by mgoltzsche** [comment]
I don't think adding a field to the generator config / `generator.yaml` to specify a custom `repositories.yaml` location would be acceptable since it would conflict with the requirement to have an allowlist of repositories a kustomization/chart is allowed to use (which is a Helm built-in concept I wouldn't weaken).

**Comment by jmorsecode** [comment]
@mgoltzsche Thank you for the prompt replies!   The generated `repositories.yaml`  via ksops encrypted values is something I briefly considered. I have some light concern if it will play nice when run within a system like ArgoCD, but will just have to play around with it.  IMO "ideally" it would be most consistent and intuitive to be able to simply set something like `helm_username` and `helm_password` at the same level of the of the repo in question requiring auth.
